ecs-init/engine/engine.go

```diff
@@ -168,7 +170,8 @@ func (e *Engine) load(image io.ReadCloser, err error) error {
 // StartSupervised starts the ECS Agent and ensures it stays running, except for terminal errors (indicated by an agent exit code of 5)
 func (e *Engine) StartSupervised() error {
 	agentExitCode := -1
-	for agentExitCode != terminalSuccessAgentExitCode && agentExitCode != terminalFailureAgentExitCode {
+	backoff0 := backoff.NewBackoff(time.Millisecond*500, time.Second*15, 0.10, 2.0, 99999999)
```

Comment:
The intent would be clearer with some constants to sweep away the magic numbers, consider something like:

```suggestion
	retryBackoff := backoff.NewBackoff(StartupRetryJitter, config.StartupRetrySeconds, DESCRIPTIVE_CONSTANT, DESCRIPTIVE_CONSTANT, DESCRIPTIVE_CONSTANT)
```

Allowing the user to passthrough some of these values could mean that in some cases system administrators could influence the unit's restart/retry policy, like on hosts running systemd.

Reply:
I think making this externally configurable is outside the scope of this change. That would also require updating documentation and messaging out the change. AFAIK there aren't any customers asking to configure the service startup retry interval of the ECS agent.

Reply:
I have changed the random numbers to named local constants so it's more clear what each argument means.

Reply:
> I think making this externally configurable is outside the scope of this change. That would also require updating documentation and messaging out the change. AFAIK there aren't any customers asking to configure the service startup retry interval of the ECS agent.

I'm thinking about this more in the context of the packaging used on Amazon Linux where the init daemon can be entrusted to handle the retry cooperatively with this code path. This would allow for the systemd unit to specify a retry policy where the host would restart `ecs-init` when appropriate and it itself be the place for configuration.

The time to add configurables is when the code is added IMO. Not going to block the merge on it though.

A remaining nit: let's change `backoff0`, I proposed `retryBackoff`, what do you think about that?

Reply:
I'm not really a fan of the retry logic living in ecs-init rather than the system's init process, but that's what we have already and I think trying to break that apart is a larger and much riskier change.

..but I will change the variable name to retryBackoff 👍